src/pybind/mgr/cephadm/module.py

```diff
@@ -2609,6 +2609,16 @@ def _config_rgw(self, spec):
             'name': 'rgw_realm',
             'value': spec.rgw_realm,
         })
+        if spec.ssl:
+            v = 'beast ssl_port=%d' % spec.get_port()
+        else:
+            v = 'beast port=%d' % spec.get_port()
+        ret, out, err = self.mon_command({
+            'prefix': 'config set',
+            'who': 'client.rgw.' + spec.service_id,
```

Comment:
Current it is "$realm.$zone".  My current thinking is that it should be "$realm.$zone[.$whatever]" so that you can deploy multiple clusters of rgws for the same zone (e.g., on different ports), and we require that zones and realms do not use ``.`` so that we can do
```
(realm, zone) = spec.service_id.split('.')[0:2]
```
or whatever.

Reply:
@liewegas how are different hosts handled? Eventually we can have a single host with multiple rgws (different ports), or multiple hosts, rgw per host, same port

Reply:
for a given service, there will currently be only one daemon per host.  but you could have different rgws for different services on the same host.  at the moment it's up to the user to make sure the ports don't collide or else the daemons won't start